src/funfuzz/js/compile_shell.py

```diff
@@ -73,6 +76,52 @@ def __init__(self, buildOpts, hgHash):
         self.jsMajorVersion = ''  # pylint: disable=invalid-name
         self.jsVersion = ''  # pylint: disable=invalid-name
 
+    def main(self, args=None):  # pylint: disable=missing-docstring,missing-return-doc,missing-return-type-doc
```

Comment:
Why is `main()` a regular method and not `staticmethod` or `classmethod`? Doing either lets you call it as `CompiledShell.main()` instead of needing to instantiate it without setting the attributes in `__init__`.